src/SixLabors.Fonts/GlyphSubstitutionCollection.cs

```diff
@@ -124,29 +128,58 @@ public void AddGlyph(ushort glyphId, CodePoint codePoint, TextDirection directio
         /// <param name="toIndex">The index to move to.</param>
         public void MoveGlyph(int fromIndex, int toIndex)
         {
-            GlyphShapingData data = this.GetGlyphShapingData(fromIndex);
+            if (fromIndex == toIndex)
+            {
+                return;
+            }
+
+            GlyphShapingData data = this[fromIndex];
             if (fromIndex > toIndex)
             {
-                int idx = fromIndex;
-                while (idx > toIndex)
+                // Move item to the left
```

Comment:
isn't this comment the wrong way around??
this loop moves what ever is in 0->1,1->2, 2->3 etc

Reply:
You're right! I wrote this one pretty late at night so the brain was frazzled.

Reply:
Fixed!